File: Devart/dotConnect/PostgreSQL/Samples/General/Backup/CS/RowDataReader/RowDataReaderDemo.cs
using System;
using System.Data;

namespace Samples
{
  /// <summary>
  /// Summary description for RowDataReaderDemo.
  /// </summary>
  public class RowDataReaderDemo : Demo
  {
    const string descriptionShort = "This sample project demonstrates using PgSqlDataReader class to obtain composite type values. The sample executes SQL statement at the PgSqlCommand and gets PgSqlDataReader to retrieve data.";
    const string descriptionFull = @"This sample project demonstrates using PgSqlDataReader class to obtain composite type values. The sample executes SQL statement at the PgSqlCommand and gets PgSqlDataReader to retrieve data.";

    public RowDataReaderDemo() 
      : base("RowDataReader", descriptionShort, descriptionFull) {
    }

    protected override BaseDemoControl CreateDemoControl(IDbConnection connection){

      return new RowDataReaderDemoControl((Devart.Data.PostgreSql.PgSqlConnection)connection);
    }

    public override string CreateSql {
      get {
        return @"CREATE TABLE dept_container(
  id int4,
  f_dept dept
);

INSERT INTO dept_container VALUES(1, '(10,N a m e,location)')";
      }
    }

    public override string DropSql {
      get {
        return @"DROP TABLE dept_container CASCADE";
      }
    }
  }
}

File: Devart/dotConnect/PostgreSQL/Samples/General/VB/RowDataReader/RowDataReaderDemo.vb
Imports Devart.Data.PostgreSql
Imports System
Imports System.Data

Namespace Samples
  Public Class RowDataReaderDemo
    Inherits Demo

    ' Fields
    Private Shadows Const descriptionFull As String = "This sample project demonstrates using PgSqlDataReader class to obtain composite type values. The sample executes SQL statement at the PgSqlCommand and gets PgSqlDataReader to retrieve data."
    Private Shadows Const descriptionShort As String = "This sample project demonstrates using PgSqlDataReader class to obtain composite type values. The sample executes SQL statement at the PgSqlCommand and gets PgSqlDataReader to retrieve data."

    ' Methods
    Public Sub New()
      MyBase.New("RowDataReader", descriptionShort, descriptionFull)
    End Sub

    Protected Overrides Function CreateDemoControl(ByVal connection As IDbConnection) As BaseDemoControl
      Return New RowDataReaderDemoControl(DirectCast(connection, PgSqlConnection))
    End Function

    ' Properties
    Public Overrides ReadOnly Property CreateSql() As String
      Get
        Return "CREATE TABLE dept_container(" & ChrW(13) & ChrW(10) & "  id int4," & ChrW(13) & ChrW(10) & "  f_dept dept" & ChrW(13) & ChrW(10) & ");" & ChrW(13) & ChrW(10) & ChrW(13) & ChrW(10) & "INSERT INTO test.dept_container VALUES(1, '(10,N a m e,location)')"
      End Get
    End Property

    Public Overrides ReadOnly Property DropSql() As String
      Get
        Return "DROP TABLE dept_container CASCADE"
      End Get
    End Property
  End Class
End Namespace